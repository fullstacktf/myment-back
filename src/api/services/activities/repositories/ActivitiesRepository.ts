import { Activity } from '../../../data/mongodb/ActivityModel';
import { SingleMongo } from '../../../data/mongodb/initMongo';
import { IdeaDTO } from 'IdeaDTO';

export class ActivitiesMongoRepository {
    
    private con: SingleMongo;
    private model = Activity;
    
    public constructor() {
        this.con = new SingleMongo('127.0.0.1','wizard','1234','myment');
        this.con.connection.on('error', console.error.bind(console, 'connection error:'));
        this.con.connection.once('open', function() {
            console.log('We are connected');
        });
    }
    public async getConnection() {
        return await this.con.connection;
    }
    
    public async getAll() {
        return await this.model.find({}).limit(10);
    }
    
    public async addActivities(data:IdeaDTO){
        //Check if exist
        if(await this.model.exists(data))
        return {error:'Data yet exist'}
        
        return await this.model.create(data)
    }
    public async updateActivity(data: IdeaDTO) {
        //Check if exist
        if(await this.model.exists(data))
        return {error:'Data yet exist'}
        
        
        throw new Error("Method not implemented.");
    }
    public async deleteActivities(data: IdeaDTO) {
        throw new Error("Method not implemented.");
    }
    public async findIdeas(type: string) {
    return await this.model.find({"category":type},"ideas");
    }
}
