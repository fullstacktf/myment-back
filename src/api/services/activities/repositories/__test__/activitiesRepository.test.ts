import { ActivitiesMongoRepository as ActivitiesRepository } from '../ActivitiesRepository';

describe('ActivitiesRepository', () => {
    const repo = new ActivitiesRepository();

    describe('getConnectionState', () => {
        test('Check connection to database', () => {
            return repo.getConnection().then(state => {
                expect(state.readyState).toBe(1);
            });
        });
    });
    describe('showActivities', () => {
        test('should return all activities', () => {
            return repo.getAll().then(data => {
                expect(data).toBe('');
            });
        });
        test('should return only food',() => {
            return repo.findBy('food').then(data =>{
                expect(data).toBe('');
            });
        })
    });
});
