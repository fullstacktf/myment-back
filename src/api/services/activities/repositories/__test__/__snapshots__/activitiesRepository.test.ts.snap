// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivitiesRepository showActivities should return json 1`] = `Array []`;
